base/reflection.jl

```diff
@@ -877,8 +877,11 @@ function code_lowered(@nospecialize(f), @nospecialize(t=Tuple); generated::Bool=
     end
 end
 
-isgenerated(m::Method) = isdefined(m, :generator)
-isgenerated(m::Core.MethodInstance) = isgenerated(m.def)
+function isgenerator(m::Core.MethodInstance)
+    def = m.def
+    isa(def, Method) || return false
+    return isdefined(def, :generator)
+end
```

Comment:
We usually talk about these being "is-generated" code or, perhaps more accurately, that it "has-generator"
```suggestion
isgenerated(m::Method) = isdefined(m, :generator)
isgenerated(m::Core.MethodInstance) = isgenerated(m.def::Method)
```